Add support to download all referenced packages from snapshot mirror

fmoessbauer opened the pull request:
For license clearing it is important to get the source of a package (both source and binary packages). For that, we add the subcommand "download" to download all referenced debian packages to a local directory. To avoid time consuming downloads on changes, we check upfront if we already have the package and only download if not.

Urist-McGit commented on src/debsbom/snapshot/client.py:
```diff
@@ -0,0 +1,192 @@
+
+from dataclasses import dataclass
+from typing import Generator, Type
+import requests
```
You are using the requests library here and have not added it to the dependencies.

fmoessbauer replied:
Shall we add it as optional dependency and only load the infrastructure when calling the download command, or shall I just require it?

Urist-McGit replied:
I think requests is ubiquitous enough we can just add it

Urist-McGit commented on src/debsbom/snapshot/client.py:
```diff
@@ -0,0 +1,192 @@
+        for s in data.get("result", []):
+            hash = s["hash"]
+            # TODO: this might be ambiguous if the same file is uploaded
+            # under different names
```
I think we can ignore that for now

fmoessbauer commented on src/debsbom/download/__init__.py:
```diff
@@ -0,0 +1,5 @@
+#
+# SPDX-License-Identifier: MIT
+
+from .download import PackageResolver
```
Done!

fmoessbauer commented:
@Urist-McGit I just rebased, improved the remaining "TODO" messages, updated the README and fixed a typo in the name of the `SnapshotNotFoundError` exception. By that, this one should be ready to go.
